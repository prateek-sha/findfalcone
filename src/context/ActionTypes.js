export const initialData = "INTIIAL_DATA";
export const error = "ERROR";
export const radiofalse = "DROPDOWN_UPDATE_TRUE";
export const dropdownfalse = "DROPDOWN_UPDATE_FALSE";
export const addPlanet = "ADD_PLANET";
export const addVehicle = "ADD_VEHICLE";
export const addResult = "ADD_RESULT";
export const resultError = "RESULT_ERROR";
export const reset = "RESET";
export const resetResult = "RESET_RESULT";
export const addTime = "ADD_TIME_TAKEN";
