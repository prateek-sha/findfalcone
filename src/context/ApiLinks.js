const DOMAIN_URI = "https://findfalcone.herokuapp.com/";

export const PLANETS_LINK = `${DOMAIN_URI}planets`;
export const VEHICLES_LINK = `${DOMAIN_URI}vehicles`;
export const TOKEN_LINK = `${DOMAIN_URI}token`;
export const FIND_LINK = `${DOMAIN_URI}find`;
