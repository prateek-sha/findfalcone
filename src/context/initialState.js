//Setting up initial state
export const initialState = {
  loading: true,
  error: false,
  planets: [],
  vehicles: [],
};

export const result = {
  planets: [],
  vehicles: [],
  error: false,
  status: false,
  planet: null,
  loading: true,
  time_taken: 0,
};
