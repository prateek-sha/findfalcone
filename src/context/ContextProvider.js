import React, { createContext, useReducer } from "react";
import axios from "axios";
import AppReducer from "./reducer/AppReducer";
import {
  error,
  initialData,
  dropdownfalse,
  radiofalse,
  addPlanet,
  addVehicle,
  addResult,
  resultError,
  reset,
  resetResult,
  addTime,
} from "./ActionTypes";
import { PLANETS_LINK, VEHICLES_LINK, TOKEN_LINK, FIND_LINK } from "./ApiLinks";
import { initialState, result } from "./initialState";
import ResultReducer from "./reducer/ResultReducer";

//Creating global context for calling values
export const globalContext = createContext(initialState);

export const ContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(AppReducer, initialState);
  const [resultstate, resultdispatch] = useReducer(ResultReducer, result);

  //function to set intial value
  async function setInitial() {
    try {
      const planets = await axios.get(PLANETS_LINK);
      //eslint-disable-next-line
      planets.data.map((planet) => {
        planet.select = -1;
      });
      const vehicles = await axios.get(VEHICLES_LINK);
      //eslint-disable-next-line
      vehicles.data.map((vehicle) => {
        vehicle.select = -1;
      });
      dispatch({
        type: initialData,
        payload: {
          planets: planets.data,
          vehicles: vehicles.data,
        },
      });
    } catch (err) {
      //Error handling
      dispatch({
        type: error,
      });
    }
  }

  const Radiofalse = (name, position) => {
    dispatch({
      type: radiofalse,
      payload: { name, position },
    });
  };
  const Dropdownfalse = (name, position) => {
    dispatch({
      type: dropdownfalse,
      payload: { name, position },
    });
  };

  const findFalcone = async () => {
    const headers = {
      Accept: "application/json",
    };

    const response = axios.post(TOKEN_LINK, {}, { headers: headers });
    const token = (await response).data.token;

    const data = {
      token,
      Planet_names: [...resultstate.planets],
      vehicle_names: [...resultstate.vehicles],
    };

    axios
      .post(FIND_LINK, data, { headers: headers })
      .then((response) => {
        resultdispatch({
          type: addResult,
          payload: response.data,
        });
      })
      .catch((err) =>
        resultdispatch({
          type: resultError,
        })
      );
  };

  const AddPlanet = (payload) => {
    resultdispatch({
      type: addPlanet,
      payload,
    });
  };

  const AddVehicle = (payload) => {
    {
      resultdispatch({
        type: addVehicle,
        payload,
      });
    }
  };

  const Reset = () => {
    dispatch({
      type: reset,
    });
    resultdispatch({
      type: resetResult,
    });
  };

  const AddTime = (payload) => {
    resultdispatch({
      type: addTime,
      payload,
    });
  };

  return (
    <globalContext.Provider
      value={{
        AddTime,
        Reset,
        resultstate,
        AddPlanet,
        AddVehicle,
        findFalcone,
        result,
        setInitial,
        state,
        planets: state.planets,
        vehicles: state.vehicles,
        Radiofalse,
        Dropdownfalse,
      }}
    >
      {children}
    </globalContext.Provider>
  );
};
