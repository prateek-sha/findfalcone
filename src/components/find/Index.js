import React, { useContext, useEffect } from "react";
import {
  Grid,
  Typography,
  makeStyles,
  CircularProgress,
} from "@material-ui/core";
import { globalContext } from "../../context/ContextProvider";
import { Planets } from "./planet/Planets";
import { useHistory } from "react-router-dom";
import Fade from "react-reveal/Fade";

const useStyles = makeStyles({
  root: {
    margin: "1em",
    color: "white",
    textAlign: "center",
  },
  center: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyItems: "center",
  },
  button: {
    color: "black",
    cursor: "pointer",
    display: "block",
    margin: "0 auto",
    padding: ".5em 1em .5em 1em",
    background: "#e61972",
    letterSpacing: "2px",
    transition: ".2s all ease-in-out",
    outline: "none",
    border: "1px solid rgba(0,0,0,1)",
    boxShadow: "3px 3px 1px #95a4ff, 3px 3px 1px 1px rgba(0,0,0,1)",

    "&:hover": {
      background: "rgba(0,0,0,1)",
      color: "white",
      border: "1px solid rgba(0,0,0,1)",
    },
  },
  disable: {
    color: "black",
    cursor: "pointer",
    display: "block",
    margin: "0 auto",
    padding: ".5em 1em .5em 1em",
    letterSpacing: "2px",
    transition: ".2s all ease-in-out",
    outline: "none",
    border: "1px solid rgba(0,0,0,1)",
    boxShadow: "3px 3px 1px #95a4ff, 3px 3px 1px 1px rgba(0,0,0,1)",
    opacity: 0.3,
  },
});

// Component for loading screen
const Loading = () => {
  const classes = useStyles();
  return (
    <Grid container direction="column" justify="center" alignItems="center">
      <Fade>
        <Grid item>
          <Typography variant="h4" className={classes.root}>
            Loading Please wait
          </Typography>
        </Grid>
        <Grid item>
          <CircularProgress color="secondary" />
        </Grid>
      </Fade>
    </Grid>
  );
};

// Component for error screen
const Error = () => {
  const classes = useStyles();
  return (
    <Grid>
      <Typography variant="h4" className={classes.root}>
        Error Loading the data
      </Typography>
    </Grid>
  );
};

export const Index = () => {
  const history = useHistory();
  const classes = useStyles();
  const { resultstate, setInitial, state, findFalcone } = useContext(
    globalContext
  );
  useEffect(() => {
    setInitial();
    // eslint-disable-next-line
  }, []);

  const handleResult = () => {
    findFalcone();
    history.push("/result");
  };

  return (
    <>
      <Grid className={classes.root}>
        <Fade left>
          <Typography variant="h5">
            Select planets you want to search in:
          </Typography>
        </Fade>
      </Grid>
      <Grid container alignItems="center" justify="center">
        <Grid item xs={0} sm={2}></Grid>
        <Grid className={classes.center} item xs={12} sm={8}>
          {state.error === true ? (
            <Error />
          ) : state.loading === true ? (
            <Loading />
          ) : (
            <Planets />
          )}
          <Fade>
            <Typography variant="h5" className={classes.root}>
              Time Taken: {resultstate.time_taken}
            </Typography>

            {resultstate.planets.length == 4 &&
            resultstate.vehicles.length == 4 ? (
              <button
                className={classes.button}
                disabled={false}
                onClick={handleResult}
              >
                Find Falcone
              </button>
            ) : (
              <button
                className={classes.disable}
                disabled={true}
                onClick={handleResult}
              >
                Find Falcone
              </button>
            )}
          </Fade>
        </Grid>
        <Grid item xs={0} sm={2}></Grid>
      </Grid>
    </>
  );
};
