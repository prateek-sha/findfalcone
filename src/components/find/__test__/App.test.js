import React from "react";
import { Index } from "../Index";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

import { ContextProvider } from "../../../context/ContextProvider";

it("matches snapshot loading", () => {
  const state = {
    loading: true,
    error: false,
    planets: [],
    vehicles: [],
  };
  const tree = renderer
    .create(
      <ContextProvider state={state}>
        <Index />
      </ContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot error", () => {
  const state = {
    loading: false,
    error: true,
    planets: [],
    vehicles: [],
  };
  const tree = renderer
    .create(
      <ContextProvider state={state}>
        <Index />
      </ContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
