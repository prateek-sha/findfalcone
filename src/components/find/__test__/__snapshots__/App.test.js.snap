// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot error 1`] = `
Array [
  <div
    className="MuiGrid-root makeStyles-root-1"
  >
    <div
      className="react-reveal"
      style={
        Object {
          "opacity": undefined,
        }
      }
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        Select planets you want to search in:
      </h5>
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
    />
    <div
      className="MuiGrid-root makeStyles-center-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <h4
              className="MuiTypography-root makeStyles-root-1 MuiTypography-h4"
            >
              Loading Please wait
            </h4>
          </div>
        </div>
        <div
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="react-reveal"
        style={
          Object {
            "opacity": undefined,
          }
        }
      >
        <h5
          className="MuiTypography-root makeStyles-root-1 MuiTypography-h5"
        >
          Time Taken: 
          0
        </h5>
      </div>
      <button
        className="react-reveal makeStyles-disable-4"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "opacity": undefined,
          }
        }
      >
        Find Falcone
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
    />
  </div>,
]
`;

exports[`matches snapshot loading 1`] = `
Array [
  <div
    className="MuiGrid-root makeStyles-root-1"
  >
    <div
      className="react-reveal"
      style={
        Object {
          "opacity": undefined,
        }
      }
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        Select planets you want to search in:
      </h5>
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
    />
    <div
      className="MuiGrid-root makeStyles-center-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <h4
              className="MuiTypography-root makeStyles-root-1 MuiTypography-h4"
            >
              Loading Please wait
            </h4>
          </div>
        </div>
        <div
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="react-reveal"
        style={
          Object {
            "opacity": undefined,
          }
        }
      >
        <h5
          className="MuiTypography-root makeStyles-root-1 MuiTypography-h5"
        >
          Time Taken: 
          0
        </h5>
      </div>
      <button
        className="react-reveal makeStyles-disable-4"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "opacity": undefined,
          }
        }
      >
        Find Falcone
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
    />
  </div>,
]
`;
